src/components/PrerequisiteTree.vue

```diff
@@ -1,89 +1,65 @@
 <template>
   <!-- not completed -->
-  <h1
-    class="text-3xl font-extrabold dark:text-gray-300 py-5 text-gray-600 mb-10"
-  >
+  <h1 class="text-3xl font-extrabold dark:text-gray-300 py-5 text-gray-600 mb-10">
     Pre-requisite Tree
   </h1>
   <div class="w-full overflow-x-auto">
-    <div
-      class="flex justify-between text-center grid-cols-5 text-black dark:text-gray-300"
-    >
-      <div
-        class="col-span-1 z-10 sm:p-3 p-1.5 rounded-lg border-blue-300 border-2 my-auto"
-      >
+    <div class="flex justify-between text-center grid-cols-5 text-black dark:text-gray-300">
+      <div class="col-span-1 z-10 sm:p-3 p-1.5 rounded-lg border-blue-300 border-2 my-auto">
         <h5 class="text-sm font-bold dark:text-white pb-2 px-2">
           Prerequisites
         </h5>
         <!-- displaying mods -->
         <div v-if="{ isOr }" class="flex-row flex space-x-2">
-          <button
-            v-for="mod in computedPreReqArr"
-            type="button"
-            class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-3 py-2 sm:px-5 xs:py-2.5 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800 mx-auto"
-          >
-            <router-link :to="'/modules/' + mod">
+          <button v-for="mod in computedPreReqArr" type="button"
+            class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-3 py-2 sm:px-5 xs:py-2.5 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800 mx-auto">
+            <!-- <router-link :to="'/module/' + mod"> -->
+            <a :href="'/module/' + mod">
               {{ mod }}
-            </router-link>
+            </a>
+            <!-- </router-link> -->
           </button>
         </div>
         <div v-else class="flex-col flex space-y-2">
-          <button
-            v-for="mod in computedPreReqArr"
-            type="button"
-            class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-3 py-2 sm:px-5 xs:py-2.5 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800"
-          >
-            <router-link :to="'/modules/' + mod">
+          <button v-for="mod in computedPreReqArr" type="button"
+            class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-3 py-2 sm:px-5 xs:py-2.5 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800">
+            <!-- <router-link :to="'/module/' + mod"> -->
+            <a :href="'/module/' + mod">
               {{ mod }}
-            </router-link>
+            </a>
```

Comment:
noticed that you replaced `<router-link>` with regular `<a>` tags instead, was that causing this issue?

ideally we should use `<router-link>` for internal redirects to prevent any unexpected behavior if the router settings were to be tweaked

Reply:
router-link was causing the issue, it was resolved when i used <a> tags. are there any other ways to go about it then?

Reply:
the issue could be caused by `<router-link>` linking to `/modules/:id` instead of `/module/:id`

this seems to be an intended change on the `router/index.js`. could you help to update all the `/module/:id` routes to `/modules/:id` instead?